api/src/services/accesscontrols.js

```diff
@@ -72,16 +80,39 @@ const grantsObject = {
     projects: {
       'read:own': ['*', '!users'], // cannot read associated users to the project
     },
+    datasets: {
+      'create:any': ['*'],
+      'read:own': ['*'],
+      'update:own': ['*'],
+    },
+    dataset_name: {
+      'read:any': ['*'],
+    },
     project_dataset_files: {
       'read:own': ['*'],
     },
     workflow: {
-      'create:any': ['stage'], // can only create a stage workflow
+      // user role can only create these four workflows
+      'create:any': [
+        CONSTANTS.WORKFLOWS.INTEGRATED,
+        CONSTANTS.WORKFLOWS.STAGE,
+        CONSTANTS.WORKFLOWS.PROCESS_DATASET_UPLOAD,
+        CONSTANTS.WORKFLOWS.CANCEL_DATASET_UPLOAD,
+      ],
+    },
+    instruments: {
+      'read:any': ['*'],
     },
     statistics: {
       'create:any': ['*'],
       'read:any': ['*'],
     },
+    upload: {
+      'create:any': ['*'],
+    },
+    fs: {
+      'read:any': ['*'],
```

Comment:
Allow user to read Filesystem contents, so that they can register datasets via the `Ingest` feature.